From 085e1ce43538b73f698dba441bec63b6b9aceb55 Mon Sep 17 00:00:00 2001
From: Romain Rigaux <romain@cloudera.com>
Date: Tue, 17 Mar 2015 20:08:01 -0700
Subject: [PATCH 1111/1173] [oozie] Generate Workflow credentials properties

If some actions are using credentials, generate their corresponding
credentials data in the workflow.
---
 apps/oozie/src/oozie/models2.py                    |    5 ++++-
 .../oozie/templates/editor2/gen/workflow.xml.mako  |    4 ++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/apps/oozie/src/oozie/models2.py b/apps/oozie/src/oozie/models2.py
index 2e91bde..0c29f96 100644
--- a/apps/oozie/src/oozie/models2.py
+++ b/apps/oozie/src/oozie/models2.py
@@ -123,7 +123,6 @@ class Workflow(Job):
                   "job_xml": "",
                   "sla_enabled": False,
                   "schema_version": "uri:oozie:workflow:0.5",
-                  "credentials": [],
                   "properties": [],
                   "sla": Workflow.SLA_DEFAULT,
                   "show_arrows": True,
@@ -229,6 +228,10 @@ class Workflow(Job):
     return self.sla_enabled or any([node.sla_enabled for node in self.nodes])
 
   @property
+  def credentials(self):
+    return list(set([cred for node in self.nodes for cred in node.data['properties']['credentials']]))
+
+  @property
   def sla(self):
     _data = self.get_data()
     return _data['workflow']['properties']['sla']
diff --git a/apps/oozie/src/oozie/templates/editor2/gen/workflow.xml.mako b/apps/oozie/src/oozie/templates/editor2/gen/workflow.xml.mako
index 713cb46..25136ca 100644
--- a/apps/oozie/src/oozie/templates/editor2/gen/workflow.xml.mako
+++ b/apps/oozie/src/oozie/templates/editor2/gen/workflow.xml.mako
@@ -29,9 +29,9 @@
     % endif
   </global>
   % endif
-  % if workflow['properties']['credentials']:
+  % if wf.credentials:
   <credentials>
-    % for cred_type in workflow['properties']['credentials']:
+    % for cred_type in wf.credentials:
     <%
       credential = mapping['credentials'][cred_type]
     %>
-- 
1.7.9.5

